addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/automation/actions/RemoveScriptAction.java

```diff
@@ -0,0 +1,121 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.scripts.automation.actions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.addon.automation.AutomationEnvironment;
+import org.zaproxy.addon.automation.AutomationProgress;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+import org.zaproxy.zap.extension.scripts.automation.ScriptJobParameters;
+import org.zaproxy.zap.extension.scripts.automation.ui.ScriptJobDialog;
+
+public class RemoveScriptAction extends ScriptAction {
+
+    public static final String NAME = "remove";
+    private static final List<String> DISABLED_FIELDS =
+            Arrays.asList(ScriptJobDialog.SCRIPT_ENGINE_PARAM, ScriptJobDialog.SCRIPT_FILE_PARAM);
+
+    public RemoveScriptAction(ScriptJobParameters parameters) {
+        super(parameters);
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getSummary() {
+        return Constant.messages.getString(
+                "scripts.automation.dialog.summary.remove", parameters.getName());
+    }
+
+    @Override
+    public List<String> verifyParameters(
+            String jobName, ScriptJobParameters params, AutomationProgress progress) {
+        List<String> list = new ArrayList<>();
+        String issue;
+        String scriptType = params.getType();
+
+        if (scriptType == null) {
+            issue =
+                    Constant.messages.getString(
+                            "scripts.automation.error.scriptTypeIsNull", jobName);
+            list.add(issue);
+            if (progress != null) {
+                progress.error(issue);
+            }
+        } else if (!this.isScriptTypeSupported(params.getType())) {
+            issue =
+                    Constant.messages.getString(
+                            "scripts.automation.error.scriptTypeNotSupported",
+                            jobName,
+                            scriptType,
+                            getName(),
+                            String.join(", ", getSupportedScriptTypes()));
+            list.add(issue);
+            if (progress != null) {
+                progress.error(issue);
+            }
+        }
+        // Note dont warn/error if script not currently in ZAP - it might be added by another job
```

Comment:
Is it worth some sort of INFO message to keep users from foot gunning?

Reply:
Same in the run class

Reply:
It will error if you run the job and the script is not found.
I _think_ thats sufficient..